Expensify/App
# Make sure that original API command promise callbacks run after authToken is refreshed

cc @tgolen @cead22 

### Details
This PR addresses an issue where authTokens are refreshed, but the API command that triggered the refresh loses it relationship to the `.then()` callback of the original API command caller.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/149348

### Tests
1. Set up authToken with a 5 second expiry or manually edit the localStorage to invalidate them ([Stack Overflow post here](https://stackoverflow.com/c/expensify/questions/7826/7827#7827)).
2. Minimize the window and maximize it again to trigger the reconnection callbacks
3. Verify that the `Authenticate` command was called and was followed by successful calls to `Get rvl=chatList` and `Report_GetHistory` in the Network tab of Chrome Dev Tools

![Screen Shot 2021-01-06 at 5 01 47 PM](https://user-images.githubusercontent.com/32969087/103846806-1200b800-5043-11eb-94de-a6124f197964.png)
![Screen Shot 2021-01-06 at 5 01 34 PM](https://user-images.githubusercontent.com/32969087/103846812-15943f00-5043-11eb-9f54-9bdf4047502a.png)
 
### Tested On


### Screenshots
<Add any necessary screenshots for all platforms if your change added or updated UI>

#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
